x/gamm/types/route.go

```diff
@@ -27,6 +27,30 @@ func (routes SwapAmountInRoutes) Validate() error {
 	return nil
 }
 
+func (routes SwapAmountInRoutes) IntermediateDenoms() []string {
+	if len(routes) < 2 {
+		return nil
+	}
+	intermediateDenoms := make([]string, 0, len(routes)-1)
+	for _, route := range routes[:len(routes)-1] {
```

Comment:
for my learning, why specify routes[:len(routes)-1] when we determined it has to be < 2 already?

Reply:
ah this is because we don't want the last token out, only wanted the intermediate one